### ISSUE TITLE
LazyCell and interior mutable

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust-clippy#10372]
### Summary

While experimenting with `LazyCell` clippy is reporting `declare_interior_mutable_const` while declaring the const item and `borrow_interior_mutable_const` while using it.

IIUC `LazyCell` is somewhat a replacement for `lazy_static` but currently unstable and behind a feature flag.

I would understand that a possible solution is postponed, until `LazyCell` is made more stable-ish, because it could change it's signature, behavior or ...

This somewhat reminds me of https://github.com/rust-lang/rust-clippy/pull/9246 which means it could easily be solved by adding a check for the `LazyCell` type, or am I wrong?

### Lint Name

declare_interior_mutable_const
borrow_interior_mutable_const

### Reproducer

I tried this code:

```rust
#![feature(once_cell)]
use std::cell::LazyCell;

const VEC: LazyCell<Vec<u32>> = LazyCell::new(|| vec![1, 2, 3]);

fn main() {
    let _ = VEC.iter();
}
```

I saw this happen:

```
warning: a `const` item should never be interior mutable
 --> src/main.rs:4:1
  |
4 | const VEC: LazyCell<Vec<u32>> = LazyCell::new(|| vec![1, 2, 3]);
  | -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | make this a static item (maybe with lazy_static)
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#declare_interior_mutable_const
  = note: `#[warn(clippy::declare_interior_mutable_const)]` on by default

warning: a `const` item with interior mutability should not be borrowed
 --> src/main.rs:7:13
  |
7 |     let _ = VEC.iter();
  |             ^^^
  |
  = help: assign this const to a local or static variable, and use the variable here
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#borrow_interior_mutable_const
  = note: `#[warn(clippy::borrow_interior_mutable_const)]` on by default
```

I expected to see this happen:

No output

### Version

```text
rustc 1.69.0-nightly (4507fdaaa 2023-02-18)
binary: rustc
commit-hash: 4507fdaaa27ea2fb59a41df2ce7d1f290da53dae
commit-date: 2023-02-18
host: x86_64-unknown-linux-gnu
release: 1.69.0-nightly
LLVM version: 15.0.7
```

**Comment by Alexendoo** [comment]
As it's a `const` it would be like having a fresh `LazyCell` at each usage site - https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=959dea0d295d45dcc8b3545d00ccc283

It'd need to be a `static` to get the expected behaviour

**Comment by hellow554** [comment]
But it can't be static, I need the [`LazyLock`](https://doc.rust-lang.org/std/sync/struct.LazyLock.html) for that.

But thanks for pointing it out!

I think I close this then :)
